src/target/rp.c

```diff
@@ -183,17 +233,18 @@ static void rp_add_flash(target *t)
 		spi_parameters.capacity = rp_get_flash_length(t);
 		spi_parameters.sector_erase_opcode = SPI_FLASH_CMD_SECTOR_ERASE;
 	}
-	rp_flash_resume(t);
 
-	DEBUG_INFO("Flash size: %zu MB\n", spi_parameters.capacity / (1024U * 1024U));
+	rp_flash_enter_xip(t);
+
+	DEBUG_INFO("Flash size: %" PRIu16 " MB\n", spi_parameters.capacity / (1024U * 1024U));
```

Comment:
PRIu16 doesn't work on ARM newlib, please use `%u` here. Also, we own this one but `MB` => `MiB` as this is using IEC not SI units, so we can't use the SI unit bases as that's wrong. Might as well also remove the extra space between the format string and units so it doesn't read so weirdly.

Reply:
I agree with `MiB` :smile: 

As for `PRIu16`, I'm testing compiling for `native` and `PRIu16` is working exactly as expected when compiling with `arm-none-eabi-gcc-10.2.1`. `arm-none-eabi-gcc-10.2.1` is compiled with `newlib-3.3.0`:
https://sourceware.org/git/?p=newlib-cygwin.git;a=blob;f=newlib/libc/include/inttypes.h;h=417ff224cb1f347ab6438a4ba75421bbe1ca85d7;hb=26894de91d296af502b53c243e71f88c06ea1a3d#l119
So I'm not too sure what you mean by:
> PRIu16 doesn't work on ARM newlib

Reply:
Ok, fixed!